server/service/frontend.go

```diff
@@ -161,6 +184,86 @@ func generateEnrollOTAURL(fleetURL string, enrollSecret string) (string, error)
 	return enrollURL.String(), nil
 }
 
+func renderEnrollPage(w io.Writer, appCfg *fleet.AppConfig, urlPrefix, enrollSecret, errorMessage string) error {
+	fs := newBinaryFileSystem("/frontend")
+	file, err := fs.Open("templates/enroll-ota.html")
+	if err != nil {
+		return fmt.Errorf("load enroll ota template: %w", err)
+	}
+
+	data, err := io.ReadAll(file)
+	if err != nil {
+		return fmt.Errorf("read bindata file: %w", err)
+	}
+
+	t, err := template.New("enroll-ota").Parse(string(data))
+	if err != nil {
+		return fmt.Errorf("create react template: %w", err)
+	}
+
+	enrollURL, err := generateEnrollOTAURL(urlPrefix, enrollSecret)
+	if err != nil {
+		return fmt.Errorf("generate enroll ota url: %w", err)
+	}
+	if err := t.Execute(w, struct {
+		EnrollURL             string
+		URLPrefix             string
+		ErrorMessage          string
+		AndroidMDMEnabled     bool
+		MacMDMEnabled         bool
+		AndroidFeatureEnabled bool
+	}{
+		URLPrefix:             urlPrefix,
+		EnrollURL:             enrollURL,
+		ErrorMessage:          errorMessage,
+		AndroidMDMEnabled:     appCfg.MDM.AndroidEnabledAndConfigured,
+		MacMDMEnabled:         appCfg.MDM.EnabledAndConfigured,
+		AndroidFeatureEnabled: true,
+	}); err != nil {
+		return fmt.Errorf("execute react template: %w", err)
+	}
+	return nil
+}
+
+func requiresEnrollOTAAuthentication(ctx context.Context, ds fleet.Datastore, enrollSecret string, noTeamIdPEnabled bool) (bool, error) {
```

Comment:
How are we going to handle calling this from the android package? Until it gets merged together into the main datastore #31218 this will be hard to call, as the android datastore does not implement all functions dictated in the full datastore.

Reply:
Unfortunately not, I tried that.

The error is as follows:
<img width="858" height="75" alt="image" src="https://github.com/user-attachments/assets/64673619-c6c1-4161-a81f-284bb8b52723" />